### ISSUE TITLE
Smithing is broken on 1.2

### ISSUE THREAD

**Issue by Vispilio** [mipen/BannerlordTweaks#67]
this mod is not compatible with smithing changes brought with the beta 1.2 branch of Banner Lords, causing instant crashes when entering smithy...

**Comment by bonerstorm69** [comment]
Confirmed instant crash on entering smithy.  Happens even if smithing changes are turned off and all other mods are turned off too.

**Comment by bonerstorm69** [comment]
Also I unblocked everything using powershell and always run as admin.

I confirmed that stopping the mod stops the issue.  Will keep it on because I love the tweaks and I hate crafting anyway.

**Comment by bonerstorm69** [comment]
LIES!  I figured out it was the "Hide Locked items" option that caused the crash. Without it on, every other smithing option works fine.

**Comment by vonenasty86** [comment]
So I tried that fix I even deleted every file related the bannerlord tweaks and reinstalled them then manually changed the JSON file in the configs folding for the game in "My Documents" and I still get the instant crash. no matter what fixes I try it still crashes on entry. I have confirmed that no matter what setting I have enabled or disabled the game still crashes. I even have a separate mod I used to use that is a stand alone hide unknown parts that I reinstalled and tried again. with bannerlord tweaks off no crash and all unknowns hidden perfectly, with bannerlord tweaks = instant crash. if anyone else has this issue let me know.

**Comment by bonerstorm69** [comment]
And you definitely unchecked "Hide Locked Items" in the mod options?  I only say that - not to frustrate you or talk down to you - because I felt like a total idiot when I turned off everything in the "crafting options" but didn't notice the "smithing options" all the way down.

Regardless, good luck bro.

**Comment by vonenasty86** [comment]
Yes, sadly. And no fear of talking down I ain't easily offended and deal
with nonsense like this a lot, but yes I went down to the smithing options
and I unchecked both of them. I even after posting deleted all bannerlord
tweaks files and started from scratch. Same thing.

I even made sure that the settings file was correct. I went to "my
documents" then into bannerlord
folder>configs>modsettings>zzbannerlordtweaks, and I opened the
bannerlordtweakssettings.JSON file which is where the settings you change
in-game are saved and why you have to close down and reboot each time you
make a change. And I made sure that file said "false" hiding locked items.
It's baffling so I am just doing without smithing for now I guess.

**Comment by RyanWilson-byte** [comment]
Just thought i'd contribute to the discussion on here that upon reading the mod option fix, to uncheck "Hide Locked Items" I tried this and only this and Smithing starting working again. 

This was a new save I made today with mods I had just installed, but I doubt that has anything to do with. I suppose it means it's "fresh" though. Everything is default for me basically. 

A temporary fix would be to make that option not default. 
But obviously it would be nice to have it enabled for those who want it. I don't think i'll use it either way, locking items is seemingly useless right now in the base game, i'm sure that's going to change in the future so we'll be prone to more bugs like this.

**Comment by vonenasty86** [comment]
Yeah I agree I don't really need that feature. I just stash anything I
truly want to keep. It's all the other smithing feature I miss. I got to
the point on one character I could make some sweet swords. Now if I want to
make an awesome sword. I gotta log. Disable banner tweaks, then log back in
Smith, then log, then log back in with banner tweaks active... Tedious and
inconvenient but oh well.

**Comment by MarcLacharite** [comment]
Pull request from joshimoo might be fixing the issue.

https://github.com/mipen/BannerlordTweaks/pull/61

**Comment by joshimoo** [comment]
Yeah the patch does fix crafting but I am currently on the 1.2 beta branch and have not tested it below that. Will update it on the weekend when taleworlds releases the office 1.2.
Hopefully @mipen will then merge it?

**Comment by Spar10Tech** [comment]
Not sure if this was reported yet but it looks like either the smithing code or experience code is causing a max increase in smithing skill when refining steel and or iron.  After the exp is gained you have max smithing skill (level 1023) with an extra skill point that can't be assigned.

I don't have either the parts or hide options checked.  Only have extra stamina.  

Game Version e1.2.1.227199 (Current public build, not dev branch)

**Comment by joshimoo** [comment]
@Spar10Tech it's unrelated to the smithing patch and probably related to the experience patch.

**Comment by Spar10Tech** [comment]
I was able to verify that with all the experience options disabled in the config the max skill event does not happen.

Also just wanted to mention the smithing parts learning seems to have increased in the vanilla build as you work with better materials.

**Comment by vonenasty86** [comment]
Yeah they really boosted smithing learning. Smithing is so broken though. I
can now craft swords that are mediocre and have negative ticks because my
smithing is like 30 and the skill required is like 90 and still sell them
for 4-6k easily. All it takes is a one/two hand sword that has long reach
and high dmg and you basically have infinite money. Only takes a couple
steel and a few iron and boom 5k... Then in late game I can make swords
with fine steel that sell easily in the 30k range.

**Comment by joshimoo** [comment]
You should try polearms, warsword something (tier 5) blade + tier 4 stick (regular hardwoord), 65k easy money, 180 dmg, 165 distance. You feel like a death reapper with the massive scythe :D

**Comment by vonenasty86** [comment]
Lol yeah I had a mod that expanded the crafting options on one game, it was
called more smithing parts or something and I made these javelins that were
130 long, 128 Pierce dmg, and 29 speed. It was so broken. They took 3 fine
steel and 1 wood... Sold for 69k.... One shot kills no matter where you hit
someone. I had to get rid of that mod it broke the game for me.

**Comment by Strobelit** [comment]
Hey. I tried the untick-smelting-option fix and it didn't work on a save file I started today.

**Comment by bonerstorm69** [comment]
Dammit.  As of today's update from TW, the "Unlock parts from smelted weapons" option also causes Smithy to freeze up.  No crash after unchecking that option. This is in addition to the freeze from "Hide locked items."

That's sad... it was one of my favorite tweaks. Do the devs do posts anywhere about these stealth updates???

**Comment by Strobelit** [comment]
I tried the unchecking the "Unlock parts from smelted weapons" box and that didn't work for me. But to be honest I'm not that fussed about troubleshooting or getting it working, I think I've played enough M&B recently.

**Comment by vonenasty86** [comment]
Lol yeah I feel you. I dropped like 90 hours on it... Now I can take a
break, read the patch notes to stay updated on content etc... And jump back
in later this summer (or winter for you southern hemispherians).

**Comment by bonerstorm69** [comment]
Aaaand I've got the 1.1.7 version of BannerLord tweaks but it still crashes on entering smithy with any smelting options enabled =_(

**Comment by Imsvale** [comment]
Unticking both options as mentioned fixed the crash for me on e1.2.1 (BLT 1.1.7). Might have been just one of them that caused it, but I unticked both just because.

**Comment by joshimoo** [comment]
Please update to 1.3.4 if you are on bannerlord 1.3
We will shortly release 1.4.0 for the new bannerlord 1.4 beta :)
The smithing tweaks should be working correctly in both 1.3.4 or 1.4.
